File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map <int, int> mp;
        int ans = -1;
        for (auto i : arr) mp[i]++;
        for (auto i : mp) {
            if (i.first == i.second) ans = max(ans, i.first);
        }
        return ans;
    }
};

File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        mp, maxi = {}, -1

        for i in arr:
            if i in mp: mp[i] += 1
            else: mp[i] = 1

        for num, x in mp.items():
            if num == x: maxi = max(maxi, num)

        return maxi